### ISSUE TITLE
Help browser not showing deep children vocabularies on Windows 7

### ISSUE THREAD

**Issue by chunes3** [factor/factor#2313]
In the Factor 0.99 2020-01-23 build, it works fine. Here's an image of the ui.gadgets vocabulary toward the bottom where it shows all of the children vocabularies:

![2020-01-23](https://user-images.githubusercontent.com/63140551/85798359-f23b1f80-b702-11ea-9124-d00e40c3fb50.png)

However, in the 0.99 2020-03-02 build, (and subsequent builds including 2020-06-23), it looks like this:

![2020-03-02](https://user-images.githubusercontent.com/63140551/85798438-1a2a8300-b703-11ea-8f05-5ae537b40c61.png)
It shows children vocabs that are one level deep such as `ui.gadgets` or `sequences.extras`. It's when you go even deeper that it seems to be a problem, like `ui.gadgets.buttons` or `io.directories.search`.

The vocabs definitely _exist_, and can be reached with e.g. `"ui.gadgets.buttons" about`.

**Comment by erg** [comment]
Looks like it regressed in 1a1ee4b4a84f60d881c16728a6d7b43cd8e8d2ed?

**Comment by erg** [comment]
IN: vocabs.hierarchy

```factor
: visible-dirs ( seq -- seq' )
    [
        {
            [ directory? ]
            [ name>> "." head? not ]
        } 1&&
    ] filter ;

: vocab-subdirs ( dir -- dirs )
    directory-entries visible-dirs [ name>> ] map! natural-sort ;

: vocab-dir? ( root name -- ? )
    over
    [ ".factor" append-vocab-dir append-path exists? ]
    [ 2drop f ]
    if ;

: (disk-vocab-children) ( root prefix -- vocabs )
    check-vocab-name
    [ vocab-dir append-path dup exists? [ vocab-subdirs ] [ drop { } ] if ]
    [ nip [ "." append '[ _ prepend ] map! ] unless-empty ]
    [ drop '[ _ over vocab-dir? [ >vocab-link ] [ <vocab-prefix> ] if ] map! ]
    2tri ;

: disk-vocabs-for-prefix ( prefix -- assoc )
    [ [ vocab-roots get ] dip '[ dup _ (disk-vocab-children) ] { } map>assoc ]
    [ unrooted-disk-vocabs [ lookup-vocab ] map! f swap 2array ]
    bi suffix ;
```

```factor
"ui.gadgets" about
```

**Comment by mrjbq7** [comment]
Fixed in d8f7bd067d49a23c891b10bfa9028ca57fb6f8a1.
